### DISCUSSION TITLE
Running dual extruders - brainstorming, and failed good idea

### DISCUSSION THREAD

**Discussion by Kremnari** [The-300lb-Gorilla/Ender-2-Pro-v427#7] [Q&A]
Thank you 300lb Gorilla for your work!

I had a great idea to use a relay to swap between two extruder motors so I can use a 2 to 1 hotend.  My plan was to use one of the part fan headers to trigger the relay.

The Failure - apparently the two part fan headers are electrically the same, and run on `M106 P0.`  I was hoping to use `M106 P1 S255` to trigger the relay.

I have two other thoughts, before resorting to another mainboard.  First, is just to run `M600` and manually hit a switch for the relay.  Second, I could throw in a microprocessor and count buzzer noises.

I'm curious if you might know of another possibility for a gcode-able voltage trigger.

**Comment by The-300lb-Gorilla** [comment]
There is an option in Marlin to run dual extruders. 
Questions?
1. Are you running the dual extruder motors on different motor ports?
2. Are you using a BL-Touch System?
3. Have you defined the number of extruders in Marlin (there are a couple of placed this needs to be done)?
4. Is this Cyclops Hot End Setup? (Example)
// For Cyclops or any "multi-extruder" that shares a single nozzle.
//#define SINGLENOZZLE
5. Are you trying to trigger a second motor and turn off the first motor with a relay because you do not have a spare motor port?

**Reply by Kremnari** [reply]
No, No, No, No, Yes!

For #3, I have not started working with my own firmware compiles quite yet.  I haven't even researched the workflow for it.

#4, The hotend I found is a single output type, so no pesky alignment issues.

#5, that is correct.  I could just use a DPDT switch to make the it simpler, but I was hoping for an automated solution.

**Comment by The-300lb-Gorilla** [comment]
I have never heard of an idea like this. A relay to run two motors. I am sorry to say, from what I know of Marlin, and that is quite a bit, I have been pouring over marlin since you first left this message. I really cannot find a viable solution for it in Marlin. BUT.... I have found a couple of videos to point you in the right direction for a two in one out extruder. It may or may not be what you are looking for. I am sorry I cannot be of more help. Here is one video that might help, but it would mean getting another mainboard:
https://www.youtube.com/watch?v=MY4uHML5pEg       Other than this, I am really useless when it comes to dual extrusion. 
Sorry I could not be of more help.
